testsuite/tests/lib-hashtbl/htbl.ml

```diff
@@ -272,3 +274,12 @@ let () =
   let l = List.sort compare l in
   List.iter (fun (k, v) -> Printf.printf "%i,%i\n" k v) l;
   Printf.printf "%i elements\n" (Hashtbl.length h)
+
+let () =
+  let h = Hashtbl.create 16 in
+  Hashtbl.add h 0 0;
+  assert (Hashtbl.replace_mem h 0 1 = Some 0);
+  assert (Hashtbl.remove_mem h 0 = Some 1);
+  assert (Hashtbl.remove_mem h 0 = None);
+  assert (Hashtbl.replace_mem h 0 1 = None);
+  assert (Hashtbl.remove_mem h 0 = Some 1)
```

Comment:
A test when `h` contains multiple bindings with the same key would be useful as well.